lib/_http_outgoing.js

```diff
@@ -125,6 +127,19 @@ function OutgoingMessage() {
 }
 util.inherits(OutgoingMessage, Stream);
 
+Object.defineProperty(OutgoingMessage.prototype, 'socket', {
+  get: function() {
+    return this[kSocket];
+  },
+  set: function(socket) {
+    this[kSocket] = socket;
+    if (socket && socket[async_id_symbol] === undefined) {
+      socket[async_id_symbol] = async_hooks.newUid();
```

Comment:
👍 on (2)
IMHO if we can placate all the assertion while `async_hooks` are disabled that would be best.